clang/docs/checkedc/Bounds-Declaration-Checking.md

```diff
@@ -0,0 +1,42 @@
+# Bounds Declaration Checking
+
+## Checking Overview
+After each statement in the clang CFG, the inferred bounds for an expression must imply the expression's declared bounds. The algorithms for inferring and checking bounds for expressions are implemented in [SemaBounds.cpp](https://github.com/microsoft/checkedc-clang/blob/master/clang/lib/Sema/SemaBounds.cpp).
+
+## Bounds Terminology
+The bounds checking code uses the following definitions to reason about bounds expressions:
+- **Declared bounds**: The bounds that the programmer has declared for a pointer expression. For example, in `array_ptr<int> p : bounds(p, p + 1) = 0`, the declared bounds of `p` are `bounds(p, p + 1)`.
+- **Inferred bounds**: The bounds that the compiler determines for a pointer expression at a particular point in checking a statement. For example, after checking `array_ptr<int> p : bounds(p, p + 1) = 0`, the inferred bounds of `p` are `bounds(any)` (since `0` by definition has bounds of `bounds(any)`).
+
+Examples:
+```
+// p has declared bounds of bounds(p, p + i).
+// q has declared bounds of bounds(q, q + j).
+void f(array_ptr<int> p : count(i), array_ptr<int> q : count(j)) {
+  // Updating after assignment: The inferred bounds of the left-hand side
+  // variable are the bounds of the right-hand side expression.
+  // p has inferred bounds of bounds(q, q + j).
+  p = q;
+
+  // Updating after modifying a variable used in declared bounds:
+  // The value of i is lost after this assignment, so the inferred
+  // bounds of p are bounds(unknown).
+  i = 0;
+
+  // Updating after modifying a variable used in declared bounds:
```

Comment:
I understand that this section is just about terminology and examples. But, it will still be useful to a reader if this sentence is changed to something like:
`// Updating after certain invertible modifications to a variable used in declared bounds:`

And, maybe one other such example where the bounds are updated to `bounds(unknown)` when the modifying expression is too complex to handle.